java/src/main/java/com/azure/graph/bulk/impl/model/GremlinEdgeVertexInfo.java

```diff
@@ -0,0 +1,165 @@
+
+package com.azure.graph.bulk.impl.model;
+
+import com.azure.graph.bulk.impl.annotations.GremlinId;
+import org.apache.commons.lang3.reflect.FieldUtils;
+
+import java.lang.reflect.Field;
+import java.util.Objects;
+
+public class GremlinEdgeVertexInfo {
+    private String id;
+    private String label;
+    private GremlinPartitionKey partitionKey;
+
+    public GremlinEdgeVertexInfo() {
+    }
+
+    public GremlinEdgeVertexInfo(String id) {
+        this.id = id;
+    }
+
+    public GremlinEdgeVertexInfo(GremlinVertex vertex) {
+        id = vertex.getId();
+        label = vertex.getLabel();
+        partitionKey = vertex.getPartitionKey();
+    }
+
+    /**
+     * Pulls values required for a GremlinEdge document off of an Object that is defined as a GremlinVertex
+     *
+     * @param from Instance of the class annotated with @GremlinVertex to pull values from
+     * @return GremlinEdgeVertexInfo containing all the required data to successfully create a link between the
+     * Vertex and another Vertex
+     */
+    public static GremlinEdgeVertexInfo fromGremlinVertex(Object from) {
+        if (from instanceof GremlinVertex) {
+            return new GremlinEdgeVertexInfo((GremlinVertex) from);
+        }
+
+        GremlinEdgeVertexInfo result = new GremlinEdgeVertexInfo();
+
+        Class<?> clazz = from.getClass();
+
+        Class<com.azure.graph.bulk.impl.annotations.GremlinVertex> annotationClass =
+                com.azure.graph.bulk.impl.annotations.GremlinVertex.class;
+        if (!clazz.isAnnotationPresent(annotationClass)) {
+            throw new IllegalArgumentException(
+                    "Class " + clazz.getSimpleName() + " is missing GremlinVertex annotation");
+        }
+
+        result.label = clazz.getAnnotation(
+                com.azure.graph.bulk.impl.annotations.GremlinVertex.class).label();
+
+        for (Field field : FieldUtils.getAllFields(clazz)) {
+            if (field.isAnnotationPresent(GremlinId.class)) {
+                try {
+                    result.id = (String) field.get(from);
+                } catch (IllegalAccessException e) {
+                    throw new ObjectConversionException(e);
+                }
+            }
+
+            if (field.isAnnotationPresent(com.azure.graph.bulk.impl.annotations.GremlinPartitionKey.class)) {
+                try {
+                    Object rawObject = field.get(from);
+                    if (rawObject instanceof GremlinPartitionKey)
+                        result.partitionKey = (GremlinPartitionKey) rawObject;
+                    else {
+                        result.partitionKey = GremlinPartitionKey.builder().value(rawObject).build();
+                    }
+                } catch (IllegalAccessException e) {
+                    throw new ObjectConversionException(e);
+                }
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
GremlinEdgeVertexInfo isn't really complete without (id, pk, label) so if we are testing for equality it needs to include all fields.

I see that this has constructors which don't include all fields but I think this will be error prone. Can the builder instance be used will the properties are being finalized and make sure the `GremlinEdgeVertexInfo` has all required properties.

Reply:
Alternative to this is to use a validate() function and have optional equality checks on the other field (only equal if either both instances are missing the field, or if both fields have the same value for the field).

Reply:
Actually the DreamWorks implementaion this sample was created illustrate why all of the values aren't always needed. 

Their implementation is/was still using tinkerpop for non-bulk related operations. The tinkerpop library doesn't need all of those values to write it's values to Cosmos (only needs the id from each of the edges). 

I was going to force the population of those values, but in many of the situations the changes required to populate (when they weren't needed) was to great. 

I prefer the alternative approach for this reason. Still allows the usage of the domain object for the tinkerpop library, without requiring values it doesn't need. But when attempting to use the same instances for bulk, when they are needed, then we can validate.